### PR TITLE
Add UT of grafana custom plugins

### PR DESCRIPTION
**Pull request by heanlan**
1. Add UT of grafana-chord-plugin
2. Improve UT coverage of grafana-sankey-plugin
3. Add workflow to run UT and report test coverage

Signed-off-by: heanlan <hanlan@vmware.com>

### CODE REVIEW

**Review by dreamtalen — APPROVED** [review]
LGTM, just want to confirm that the huge line changes in `yarn.lock` are automatically generated and not for review right?

File: .github/workflows/typescript.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      has_changes: ${{ steps.check_diff.outputs.has_changes }}
+
+  test-unit:
+    needs: check-typescript-changes
+    if: ${{ needs.check-typescript-changes.outputs.has_changes == 'yes' }}
+    name: Unit test
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node-version: [16.x]
```
Comment (salv-orlando) [inline .github/workflows/typescript.yml:37]: We usually don't specify node-version. Do we need to pin it for some reason?

**Review by salv-orlando — COMMENTED** [review]
Just a question about the github workflow

**Comment by salv-orlando** [comment]
> 

I'm not sure a human being could review those dependency changes in yarn.lock  :)
It's generated by yarn, so there should be no need for review.

File: .github/workflows/typescript.yml (hunk 1 above)
Comment (heanlan) [reply .github/workflows/typescript.yml]: Yes, you're right. We don't need to specify the node version. Thanks.

**Comment by heanlan** [comment]
> LGTM, just want to confirm that the huge line changes in `yarn.lock` are automatically generated and not for review right?

Those are auto-generated, no need to be reviewed. In order to successfully run the test on D3 react component, I changed some dependencies in package.json, that's why the yarn.lock file changed a lot.

**Comment by dreamtalen** [comment]
> > LGTM, just want to confirm that the huge line changes in `yarn.lock` are automatically generated and not for review right?
> 
> Those are auto-generated, no need to be reviewed. In order to successfully run the test on D3 react component, I changed some dependencies in package.json, that's why the yarn.lock file changed a lot.

Got it, thanks!

**Review by yuntanghsu — APPROVED** [review]

**Pull request merged**